code/game/machinery/droneDispenser.dm

```diff
@@ -39,25 +39,29 @@
 /obj/machinery/droneDispenser/New()
 	..()
 	health = max_health
+	materials = new(src, list(MAT_METAL=1, MAT_GLASS=1), MINERAL_MATERIAL_AMOUNT*MAX_STACK_SIZE*2)
+	using_materials = list(MAT_METAL=metal_cost, MAT_GLASS=glass_cost)
 	SSmachine.processing |= src
 
 /obj/machinery/droneDispenser/Destroy()
 	SSmachine.processing -= src
```

Comment:
I'm not sure but don't we need to qdel datum references? like we do qdel(reagents) for reagent containers.